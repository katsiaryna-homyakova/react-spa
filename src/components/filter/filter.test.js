import React from 'react';
import { mount } from 'enzyme';
import Filter from './filter';

describe('filter component', () => {
  const mockClick = jest.fn(); let
    component;

  beforeAll(() => {
    const filters = [
      { id: 'releaseDate', value: 'Release date' },
      { id: 'genre', value: 'Genre' },
    ];
    component = mount(
      <Filter
        title="Sort by"
        filters={filters}
        handleChangeFilter={mockClick}
        activeFilter="someFilter"
      />,
    );
  });

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('handle click', () => {
    component
      .find('button')
      .at(1)
      .simulate('click');
    expect(mockClick).toHaveBeenCalledTimes(1);
    expect(component).toMatchSnapshot();
  });
});
