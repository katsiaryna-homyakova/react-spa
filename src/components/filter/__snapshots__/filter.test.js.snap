// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter component handle click 1`] = `ReactWrapper {}`;

exports[`filter component renders correctly 1`] = `ReactWrapper {}`;
